lib/CCreatureHandler.cpp

```diff
@@ -1322,14 +1333,20 @@ CreatureID CCreatureHandler::pickRandomMonster(CRandomGenerator & rand, int tier
 	return CreatureID(r);
 }
 
-void CCreatureHandler::addBonusForTier(int tier, std::shared_ptr<Bonus> b)
+void CCreatureHandler::addBonusForTier(int tier, const std::shared_ptr<Bonus> & b)

```

Comment:
Do you want to make const Bonus (std::shared_ptr<const Bonus>) or save some CPU to pass shared_ptr struct by ref instead of by value? Or am I missing something?

Reply:
Because it calls 'addNewBonus' which has following signature:
`void CBonusSystemNode::addNewBonus(const std::shared_ptr<Bonus>& b)`

Reply:
Ok.